core/acl/aclservice.go

```diff
@@ -0,0 +1,477 @@
+	a.accountService = app.MustComponent[account.Service](ap)
+	a.inviteStore = app.MustComponent[invitestore.Service](ap)
+	a.fileAcl = app.MustComponent[fileacl.Service](ap)
+	a.objectGetter = app.MustComponent[getblock.ObjectGetter](ap)
+	return nil
+}
+
+func (a *aclService) Name() (name string) {
+	return CName
+}
+
+func (a *aclService) Remove(ctx context.Context, spaceId string, identities []crypto.PubKey) error {
+	removeSpace, err := a.spaceService.Get(ctx, spaceId)
+	if err != nil {
+		return err
+	}
+	newPrivKey, _, err := crypto.GenerateRandomEd25519KeyPair()
+	if err != nil {
+		return err
+	}
+	cl := removeSpace.CommonSpace().AclClient()
+	err = cl.RemoveAccounts(ctx, list.AccountRemovePayload{
+		Identities: identities,
+		Change: list.ReadKeyChangePayload{
+			MetadataKey: newPrivKey,
+			ReadKey:     crypto.NewAES(),
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("%w, %w", ErrAclRequestFailed, err)
+	}
+	return nil
+}
+
+func (a *aclService) Cancel(ctx context.Context, spaceId string) (err error) {
+	// TODO: finish this by implementing space offload for join canceled spaces (?)
+	sp, err := a.spaceService.Get(ctx, spaceId)
+	if err != nil {
+		return err
+	}
+	cl := sp.CommonSpace().AclClient()
+	err = cl.CancelRequest(ctx)
+	if err != nil {
+		return fmt.Errorf("%w, %w", ErrAclRequestFailed, err)
+	}
+	return nil
+}
+
+func (a *aclService) Decline(ctx context.Context, spaceId string, identity crypto.PubKey) (err error) {
+	sp, err := a.spaceService.Get(ctx, spaceId)
+	if err != nil {
+		return err
+	}
+	cl := sp.CommonSpace().AclClient()
+	err = cl.DeclineRequest(ctx, identity)
+	if err != nil {
+		return fmt.Errorf("%w, %w", ErrAclRequestFailed, err)
+	}
+	return nil
+}
+
+func (a *aclService) ChangePermissions(ctx context.Context, spaceId string, perms []AccountPermissions) error {
+	sp, err := a.spaceService.Get(ctx, spaceId)
+	if err != nil {
+		return err
+	}
+	var listPerms []list.PermissionChangePayload
+	acl := sp.CommonSpace().Acl()
+	acl.RLock()
+	for _, perm := range perms {
+		var aclPerms list.AclPermissions
+		switch perm.Permissions {
+		case model.ParticipantPermissions_Reader:
+			aclPerms = list.AclPermissionsReader
+		case model.ParticipantPermissions_Writer:
+			aclPerms = list.AclPermissionsWriter
+		default:
+			acl.RUnlock()
+			return ErrIncorrectPermissions
+		}
+		curPerms := acl.AclState().Permissions(perm.Account)
+		if curPerms.NoPermissions() {
+			acl.RUnlock()
+			return ErrNoSuchUser
+		}
+		if curPerms == aclPerms {
+			continue
+		}
+		listPerms = append(listPerms, list.PermissionChangePayload{
+			Identity:    perm.Account,
+			Permissions: aclPerms,
+		})
+	}
+	acl.RUnlock()
+	cl := sp.CommonSpace().AclClient()
+	err = cl.ChangePermissions(ctx, list.PermissionChangesPayload{
+		Changes: listPerms,
+	})
+	if err != nil {
+		return fmt.Errorf("%w, %w", ErrAclRequestFailed, err)
+	}
+	return nil
+}
+
+func (a *aclService) Exit(ctx context.Context, spaceId string) error {
```

Comment:
Maybe Join/Leave can be a better naming pair?

Reply:
sure

Reply:
sure, but then we have to rename api method